### DISCUSSION TITLE
How to make form read only especially for upload fields

### DISCUSSION THREAD

**Discussion by maatata** [orchidsoftware/platform#1728] [Q&A]
Hi All

I have come across a situation which I need to show a form but in a view only status. No one should be able to change anything. I know I can disable the edit functionality and remove the update button on the screen but I need the fields to be actually read only too. I believe there is a method for input fields which is called `readonly` which I can use that fine. However my issue is how to make the upload fields read only and don't let the user to upload new files and just view the currently uploaded files on that page?

There is no functionality for such that thing to be able to make the upload or image or cropper field readonly and disable the upload on them.

If you could help me how to achieve this simply it would be much appreciated.

Thank you very much.

**Comment by maatata** [comment]
Any solution on this?

**Reply by tabuna** [reply]
I think the file upload box could be replaced with displaying your layout wherever you list or preview your files.

```php
use Orchid\Support\Facades\Layout;

public function layout(): array
{
    return [
        Layout::view('myTemplate'),
    ];
}
```

Also consider the possibility of displaying records, not in the same form as they are entered, but differently.
For example, using [Legend](https://orchid.software/en/docs/legend/)
